gluten-core/src/main/scala/org/apache/spark/shuffle/GlutenShuffleUtils.scala

```diff
@@ -91,6 +91,20 @@ object GlutenShuffleUtils {
     }
   }
 
+  def getCompressionBufferSize(conf: SparkConf, codec: String): Int = {
+    if ("lz4" == codec) {
+      Math.max(
+        conf.get(IO_COMPRESSION_LZ4_BLOCKSIZE).toInt,
+        GlutenConfig.GLUTEN_SHUFFLE_COMPRESSION_BUFFER_MIN_SIZE)
```

Comment:
Can we support set the config? `GLUTEN_SHUFFLE_COMPRESSION_BUFFER_MIN_SIZE`

Reply:
Looks like the default value 64 is much less than other compression kind default value 32 * 1024

Reply:
64 is not the default value, unless user set `IO_COMPRESSION_LZ4_BLOCKSIZE` to a very small size.

Reply:
WE could set a more reasonable value, maybe 32 * 1024?

Reply:
Per discussion, we will throw exception if `IO_COMPRESSION_LZ4_BLOCKSIZE` < 4. For each serialized row, the row size takes 4 bytes. Therefore 4 bytes is the minimum acceptable compression block size in Gluten.

Note here for the Spark exceptions:

- lz4: spark.io.compression.lz4.blockSize=0
```
Caused by: java.lang.IllegalArgumentException: blockSize must be >= 64, got 0
  at net.jpountz.lz4.LZ4BlockOutputStream.compressionLevel(LZ4BlockOutputStream.java:60)
  at net.jpountz.lz4.LZ4BlockOutputStream.<init>(LZ4BlockOutputStream.java:101)
  at org.apache.spark.io.LZ4CompressionCodec.compressedOutputStream(CompressionCodec.scala:151)
  at org.apache.spark.broadcast.TorrentBroadcast$.$anonfun$blockifyObject$2(TorrentBroadcast.scala:361)
  at scala.Option.map(Option.scala:230)
  at org.apache.spark.broadcast.TorrentBroadcast$.blockifyObject(TorrentBroadcast.scala:361)
  at org.apache.spark.broadcast.TorrentBroadcast.writeBlocks(TorrentBroadcast.scala:161)
  at org.apache.spark.broadcast.TorrentBroadcast.<init>(TorrentBroadcast.scala:99)
  at org.apache.spark.broadcast.TorrentBroadcastFactory.newBroadcast(TorrentBroadcastFactory.scala:38)
  at org.apache.spark.broadcast.BroadcastManager.newBroadcast(BroadcastManager.scala:78)
  at org.apache.spark.SparkContext.broadcastInternal(SparkContext.scala:1662)
  at org.apache.spark.SparkContext.broadcast(SparkContext.scala:1644)
  at org.apache.spark.scheduler.DAGScheduler.submitMissingTasks(DAGScheduler.scala:1585)
  at org.apache.spark.scheduler.DAGScheduler.submitStage(DAGScheduler.scala:1402)
  at org.apache.spark.scheduler.DAGScheduler.handleJobSubmitted(DAGScheduler.scala:1337)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3003)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
  at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
```

- zstd: spark.io.compression.zstd.bufferSize=0

```
Caused by: java.lang.IllegalArgumentException: Buffer size <= 0
  at java.io.BufferedOutputStream.<init>(BufferedOutputStream.java:74)
  at org.apache.spark.io.ZStdCompressionCodec.compressedOutputStream(CompressionCodec.scala:237)
  at org.apache.spark.broadcast.TorrentBroadcast$.$anonfun$blockifyObject$2(TorrentBroadcast.scala:361)
  at scala.Option.map(Option.scala:230)
  at org.apache.spark.broadcast.TorrentBroadcast$.blockifyObject(TorrentBroadcast.scala:361)
  at org.apache.spark.broadcast.TorrentBroadcast.writeBlocks(TorrentBroadcast.scala:161)
  at org.apache.spark.broadcast.TorrentBroadcast.<init>(TorrentBroadcast.scala:99)
  at org.apache.spark.broadcast.TorrentBroadcastFactory.newBroadcast(TorrentBroadcastFactory.scala:38)
  at org.apache.spark.broadcast.BroadcastManager.newBroadcast(BroadcastManager.scala:78)
  at org.apache.spark.SparkContext.broadcastInternal(SparkContext.scala:1662)
  at org.apache.spark.SparkContext.broadcast(SparkContext.scala:1644)
  at org.apache.spark.scheduler.DAGScheduler.submitMissingTasks(DAGScheduler.scala:1585)
  at org.apache.spark.scheduler.DAGScheduler.submitStage(DAGScheduler.scala:1402)
  at org.apache.spark.scheduler.DAGScheduler.handleJobSubmitted(DAGScheduler.scala:1337)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3003)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
  at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
```